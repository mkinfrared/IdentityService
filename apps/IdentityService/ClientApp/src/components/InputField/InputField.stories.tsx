import { Meta, Story } from "@storybook/react";

import { InputField } from "./InputField";
import { InputFieldProps } from "./InputField.type";
import css from "./Story.module.scss";

export default {
  title: "COMPONENTS/InputField",
  component: InputField,
  parameters: {
    componentSubtitle: "Subtitle goes here",
  },
} as Meta;

const Template: Story<InputFieldProps> = (args) => (
  <div className={css.Story}>
    <InputField {...args} />
  </div>
);

const Default = Template.bind({});

Default.args = {};

Default.parameters = {
  docs: {
    storyDescription: "Story description",
  },
};

export { Default };
