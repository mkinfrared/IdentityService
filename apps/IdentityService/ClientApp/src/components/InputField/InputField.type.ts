import { TextFieldProps } from "@identity-service/ui";

export type InputFieldProps = Omit<TextFieldProps, "inputRef"> & {
  /**
   * a string that will be applied as a css class to parent element
   */
  className?: string;
};
