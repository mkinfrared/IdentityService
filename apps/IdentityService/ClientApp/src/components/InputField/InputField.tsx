import { TextField } from "@identity-service/ui";
import { Ref, forwardRef, memo } from "react";

import { InputFieldProps } from "./InputField.type";

const InputField = forwardRef(
  ({ className, ...rest }: InputFieldProps, ref: Ref<HTMLInputElement>) => (
    <TextField {...rest} className={className} inputRef={ref} />
  ),
);

InputField.displayName = "InputField";

export { InputField };

export default memo(InputField);
