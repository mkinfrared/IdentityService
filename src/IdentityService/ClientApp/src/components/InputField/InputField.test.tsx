import { render } from "@testing-library/react";

import { InputField } from "./InputField";

describe("<InputField />", () => {
  const Component = <InputField />;

  it("should be defined", () => {
    expect(InputField).toBeDefined();
  });

  it("match the snapshot", () => {
    const { container } = render(Component);

    expect(container).toMatchSnapshot();
  });
});
