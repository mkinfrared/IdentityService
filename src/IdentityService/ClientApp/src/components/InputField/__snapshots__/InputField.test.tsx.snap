// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputField /> match the snapshot 1`] = `
<div>
  <label
    class="TextField-module_TextField__3-bgB"
    data-testid="TextField"
  >
    <div
      class="TextField-module_input__3FXIK"
    >
      <input
        type="text"
        value=""
      />
    </div>
    <p
      class="Text-module_Text__3EbTM TextField-module_error__3xz6q"
      data-testid="Text"
    />
  </label>
</div>
`;
